lib/twitter_web/controllers/browse_controller.ex

```diff
@@ -0,0 +1,68 @@
+defmodule TwitterWeb.BrowseController do
+  use TwitterWeb, :controller
+  alias Twitter.Twits
+  alias Twitter.Twits.Post
+  import Ecto.Query, warn: false
+  alias Twitter.Repo
+
+  plug :authenticate_user
+
+  @doc """
+  list all posts made by any user
+  """
+  @spec index(Plug.Conn.t(), map(), User.t()) :: Plug.Conn.t()
+  def index(conn, %{"author_id" => author_id}, _current_user) do
+    posts = Twits.list_posts_by_author_id(author_id)
+    render(conn, "index.html", posts: posts)
+  end
+
+  @doc """
+  show a public post /browse/author_id/post_id
+  """
+  @spec show(Plug.Conn.t(), map(), User.t()) :: Plug.Connt.t()
+  def show(conn, %{"author_id" => author_id, "id" => id}, _current_user) do
+    post = Repo.one!(from p in Post, where: p.user_id == ^author_id, where: p.id == ^id)
+    render(conn, "show.html", post: post)
+  end
+
+  @doc """
+  add like for a given post_id. Here, browse_id is the post_id
+  """
+  @spec like(Plug.Conn.t(), map(), User.t()) :: Plug.Conn.t()
+  def like(conn, %{"author_id" => author_id, "browse_id" => browse_id}, current_user) do
+    case Twits.create_like(current_user, browse_id) do
+      {:ok, _} ->
+        conn
+        |> put_flash(:info, "You liked this")
+
+      {:error, _} ->
+        conn
+        |> put_flash(:info, "Couldn't send like")
+    end
```

Comment:
This doesn't do anything I believe?
`put_flash` takes a conn and returns a new conn, but you aren't using the return value anywhere.
Elixir is a functional language with immutable data 😉
https://hexdocs.pm/phoenix/Phoenix.Controller.html#put_flash/3

Reply:
You mean I don't need the return value of conn?

Reply:
`conn |> put_flash(:info, "You liked this")` doesn't modify `conn` (it cannot, it's immutable), it returns a new `Plug.Conn`.

You can think about it this way;

```elixir
iex(1)> conn = %{}
%{}
iex(2)> Map.put(conn, :flash, "foo")
%{flash: "foo"}
iex(3)> conn
%{}
```